### ISSUE TITLE
Add differential IK to joint_state_group

### ISSUE THREAD

**Issue by sachinchitta** [moveit/moveit_core#14]
probably something along the lines of  bool setFromDiffIK(const Twist &twist, const double dT, const diffIKCallback &callback) where the diffIKCallback returns a vector of joint velocities that are to be applied in the nullspace. Then add a set of default callbacks: e.g. move towards reference joint position in nullspace or move away from joint limits, etc.

**Comment by marioprats** [comment]
I have a first working version of this. This is what I've done:
- In moveit_core, add setFromDiffIK to joint_state_group. This method basically computes qdot=Jinv*xdot. It uses the getJacobian method, so no dependencies to KDL. If a callback is given, its result (a vector of joint velocities) is projected into the nullspace of the jacobian, thus allowing defining secondary tasks. I also overloaded getVariableValues and setVariableValues for accepting an Eigen::VectorXd as argument, and avoid having to convert from a std::vector, although there is probably something for that in eigen...
- In moveit_ros, add interaction modes into robot_interaction. There is the default POSITION_IK, and the new VELOCITY_IK. If in velocity mode, setFromDiffIK is called instead of setFromIK. This allows to teleop the robot as with a cartesian controller. I have defined a default callback for the secondary task that tries to keep away from joint limits. We should probably move secondary tasks into a separate file / class structure.
- In moveit_ros, expose the previous functionality into the rviz plugin, by a new checkbox inside the Planning Request group, called "Cartesian teleop". If it is set, robot_interaction runs in VELOCITY_IK mode, thus resembling a cartesian controller  (ros-planning/moveit_ros#6). The changes also include the capability of moving the interactive markers to any place, even if IK fails (ros-planning/moveit_ros#124).

I'll submit a pull request shortly, tell me what you don't like ;)
